src/predicate/isNumber.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Checks if a given value is a number.
+ *
+ * This function can also serve as a type predicate in TypeScript, narrowing the type of the argument to `number`.
+ *
+ * @param {unknown} value The value to check if it is a number.
+ * @returns {value is number} Returns `true` if `value` is a number, else `false`.
+ *
+ * @example
+ * const value1 = 123;
+ * const value2 = 'abc';
+ * const value3 = true;
+ *
+ * console.log(isNumber(value1)); // true
+ * console.log(isNumber(value2)); // false
+ * console.log(isNumber(value3)); // false
+ */
+export function isNumber(value: unknown): value is number {
+  return typeof value === 'number' && !Number.isNaN(value);
```

Comment:
I am not sure if we should return `false` for `NaN` (`lodash` also returns `true` for `NaN`). Is there some background for checking the value is not `NaN`?

Reply:
Forgot to notice that, will align this behavior same to lodash.